internal/extproc/translator/anthropic_awsanthropic.go

```diff
@@ -0,0 +1,184 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package translator
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"maps"
+	"net/url"
+
+	"github.com/anthropics/anthropic-sdk-go"
+	extprocv3 "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+
+	anthropicschema "github.com/envoyproxy/ai-gateway/internal/apischema/anthropic"
+	"github.com/envoyproxy/ai-gateway/internal/internalapi"
+)
+
+// NewAnthropicToAWSAnthropicTranslator creates a translator for Anthropic to AWS Bedrock Anthropic format.
+// AWS Bedrock supports the native Anthropic Messages API, so this is essentially a passthrough
+// translator with AWS-specific path modifications.
+func NewAnthropicToAWSAnthropicTranslator(apiVersion string, modelNameOverride internalapi.ModelNameOverride) AnthropicMessagesTranslator {
+	return &anthropicToAWSAnthropicTranslator{
+		apiVersion:        apiVersion,
+		modelNameOverride: modelNameOverride,
+	}
+}
+
+type anthropicToAWSAnthropicTranslator struct {
+	// TODO: reuse anthropicToAnthropicTranslator and embed it here to avoid code duplication.
```

Comment:
can you resolve this TODO instead of copy-and-paste the existing especially the response handling?

Reply:
this is the first PR on this translation so can you go ahead and do the refactoring?